12.0 mig web m2x options

gustitammam opened the pull request.

Commit:
- Rename web_m2o_enhanced to web_m2x_options
- Add static/description/index.html

fix: use include instead of extend in js function inheritance.

fix: not overwriting the existing object references with the result of the include

fix: update name according to new module name.

fix: error when displaying many2many field without options defined.

hbrunn committed:
move all addons to __unported__, set all addons as not installable

Commit:
fix: obscur wrong path on last commit. recreate portage from scratch.

hbrunn committed:
support 'open' on many2many_tags

[ADD] support 'no_open_edit' on many2one
[FIX] typos

antespi committed:
web_m2x_options check Many2one and Many2many options

oca-transbot committed:
OCA Transbot updated translations from Transifex

antespi committed:
Limit name_search to 160 like Odoo core does

zakmak88 committed:
Migration of module web_m2x_options. Adapt JS file to the new JS API and make the module installable [MIG] adapt form.js to the new API. [FIX] Fix bug mentioned in pull #262. [MIG] Make the module installable.

lmignon committed:
Makes the module compatible with Odoo community

oca-transbot committed:
OCA Transbot updated translations from Transifex

zakmak88 committed:
web_m2x_options: s/ev.handleObj.selector/currentTarget * many2many with `options={'open': True}` it always open the first element with this fix it will open the right one

yajo committed:
Fix Qweb templates. (#561)

* [FIX][web_m2x_options] Fix Qweb templates.

They were trying to replace non-existing elements, and this was being logged to console:

    Can't find "a.oe_m2o_cm_button" when extending template FieldMany2One
    Can't find "span.badge" when extending template FieldMany2ManyTag

* Raise version correctly.

* [FIX] Make many2many_tags tag deletion work again

* Correct replacement of event

Instead of overwriting all events from upstream widget, better just overwrite the one you need and inherit the rest.

chienandalu committed:
[MIG] web_m2x_options: migrated to 10.0

jesusVMayor committed:
[IMP] web_m2x_options: Perf issue https://github.com/OCA/web/issues/615
* fix unecessary calls

Do check_access_rights and disable_quick_create calls only once per field.

jesusVMayor committed:
web_m2x_options: Hide the create/create and edit... button when the user does not have creation permission.

grindtildeath committed:
[10.0][ADD] option to deactivate color picker on many2many_tags (#713)

yajo committed:
[FIX] web_m2x_options: Compatibilize with web_advanced_search_x2x

`web_m2x_options` is supposed to render fields in the context of a form or tree view, where domains can safely be applied because there's a main record that includes a context.

However, when installing along with `web_advanced_search_x2x`, they produce an incompatibility when a x2x field's domain is defined and depends on the current record's context, because a search view has no notion of a *current record*.

The fix is simple: try to do as usual, and if it fails, try without the field's domain.

Without this patch, an exception like this would be raised, i.e. when both addons are installed and you are trying to search project tasks by stage:

``` Error: NameError: name 'project_id' is not defined
http://localhost/web/static/lib/py.js/lib/py.js:370# Traceback:# Changes to be committed:
PY_ensurepy@http://localhost/web/static/lib/py.js/lib/py.js:370:19# modified: static/src/js/form.js
py.evaluate@http://localhost/web/static/lib/py.js/lib/py.js:1340:20#
py.evaluate@http://localhost/web/static/lib/py.js/lib/py.js:1397:35
py.evaluate@http://localhost/web/static/lib/py.js/lib/py.js:1409:34
py.eval@http://localhost/web/static/lib/py.js/lib/py.js:1453:16
eval_domains/<@http://localhost/web/static/src/js/framework/pyeval.js:869:39
_.forEach@http://localhost/web/static/lib/underscore/underscore.js:145:9
_.mixin/</_.prototype[name]@http://localhost/web/static/lib/underscore/underscore.js:1484:29
eval_domains@http://localhost/web/static/src/js/framework/pyeval.js:860:5
eval_domains/<@http://localhost/web/static/src/js/framework/pyeval.js:873:39
_.forEach@http://localhost/web/static/lib/underscore/underscore.js:145:9
_.mixin/</_.prototype[name]@http://localhost/web/static/lib/underscore/underscore.js:1484:29
eval_domains@http://localhost/web/static/src/js/framework/pyeval.js:860:5
eval_domains/<@http://localhost/web/static/src/js/framework/pyeval.js:873:39
_.forEach@http://localhost/web/static/lib/underscore/underscore.js:145:9
_.mixin/</_.prototype[name]@http://localhost/web/static/lib/underscore/underscore.js:1484:29
eval_domains@http://localhost/web/static/src/js/framework/pyeval.js:860:5
pyeval@http://localhost/web/static/src/js/framework/pyeval.js:977:16
eval_arg@http://localhost/web/static/src/js/framework/pyeval.js:988:16
ensure_evaluated@http://localhost/web/static/src/js/framework/pyeval.js:1011:21
call@http://localhost/web/static/src/js/framework/data_model.js:56:9
name_search@http://localhost/web/static/src/js/framework/data.js:537:16
get_search_result@http://localhost/web_m2x_options/static/src/js/form.js:130:50
OdooClass.extend/Class.include/</prototype[name]</<@http://localhost/web/static/src/js/framework/class.js:122:35
source@http://localhost/web/static/src/js/views/form_relational_widgets.js:271:17
_search@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:6823:3
$.widget/</proxiedPrototype[prop]</<@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:415:19
search@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:6815:10
$.widget/</proxiedPrototype[prop]</<@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:415:19
$.widget.bridge/$.fn[name]/<@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:508:19
each@http://localhost/web/static/lib/jquery/jquery.js:383:49
each@http://localhost/web/static/lib/jquery/jquery.js:136:24
$.widget.bridge/$.fn[name]@http://localhost/web/static/lib/jquery.ui/jquery-ui.js:494:4
render_editable/<@http://localhost/web/static/src/js/views/form_relational_widgets.js:189:21
dispatch@http://localhost/web/static/lib/jquery/jquery.js:4640:50
add/elemData.handle@http://localhost/web/static/lib/jquery/jquery.js:4309:41
```

ernestotejeda committed:
[MIG] web_m2x_options: Migration to 11.0

web_m2x_options: Fix usage for non-admins

web_m2x_options: Reduce rpc calls

web_m2x_options: Update manifest and readme

oca-travis committed:
[UPD] Update web_m2x_options.pot

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 91.7% (11 of 12 strings)

Translation: web-11.0/web-11.0-web_m2x_options
Translate-URL: https://translation.odoo-community.org/projects/web-11-0/web-11-0-web_m2x_options/pt_BR/

gustitammam committed:
[MIG] web_m2x_options: Migration to 12.0

PCatinean committed:
[FIX] open option on many2many_tags

pedrobaeza commented:
Please add #1118 to this PR

nikul-serpentcs commented on web_m2x_options/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+
```
@gustitammam IMO Add copyright

gustitammam replied:
> @gustitammam IMO Add copyright

@nikul-serpentcs here: https://github.com/OCA/web/pull/1086/commits/85a462da3f372024cf8eae315484f2fda0a32013

nikul-serpentcs commented on web_m2x_options/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    'website': 'https://github.com/OCA/web',
+    'license': 'AGPL-3',
+    "depends": [
+        'base',
```
@gustitammam remove it

gustitammam replied:
> @gustitammam remove it

@nikul-serpentcs Sorry, I'm not sure which line you mentioning, did you suggest to remove `base` as the dependency ?

nikul-serpentcs replied:
Yes, `base`
because it depends on `web`.

gustitammam replied:
> Yes, `base`
> because it depends on `web`.

@nikul-serpentcs `base` removed: https://github.com/OCA/web/pull/1086/commits/89aaf7342ac3851e34563fcc121a209c91896000

nikul-serpentcs reviewed (changes requested):
Improve Code

gustitammam commented:
> Please add #1118 to this PR

@pedrobaeza This is it: https://github.com/OCA/web/pull/1086/commits/12c88b0b2dbb5a9a70029322b330580fe2a5fa91

pedrobaeza commented:
@gustitammam please don't do merge operations over the branch, as this scrambles the commit history and provoke conflicts for next versions. Please rebase now over 12.0 branch for fixing commit history.

gustitammam commented:
> @gustitammam please don't do merge operations over the branch, as this scrambles the commit history and provoke conflicts for next versions. Please rebase now over 12.0 branch for fixing commit history.

I just rebase it, @pedrobaeza please review, is this correct?

yajo committed:
[FIX] web_m2x_options: Fix error in tree views with many2many_tags

The method `_onOpenColorPicker` is defined on `FormFieldMany2ManyTags`, not in `FieldMany2ManyTags`, so it can only be called on its descendants.

Otherwise, clicking on a `many2many_tags` tag when found in a tree view produces an error such as:

    TypeError: self._onOpenColorPicker is not a function

    http://localhost/web_m2x_options/static/src/js/form.js:354
    Rastreo de error:
    _onOpenBadge@http://localhost/web_m2x_options/static/src/js/form.js:354:17
    proxy/<@http://localhost/web/static/src/js/core/mixins.js:279:20
    dispatch@http://localhost/web/static/lib/jquery/jquery.js:4640:50
    add/elemData.handle@http://localhost/web/static/lib/jquery/jquery.js:4309:41

(cherry picked from commit ddbbe81f4ea38c6685dd41efb5f11f1aeb8d0f9d)

pedrobaeza commented:
There's a new change in 11.0: https://github.com/OCA/web/commit/ddbbe81f4ea38c6685dd41efb5f11f1aeb8d0f9d

pedrobaeza commented:
@gustitammam the rest seems OK. Thanks for the efforts.

gustitammam commented:
> There's a new change in 11.0: [ddbbe81](https://github.com/OCA/web/commit/ddbbe81f4ea38c6685dd41efb5f11f1aeb8d0f9d)

Here is it: https://github.com/OCA/web/pull/1086/commits/877bc320b853a8fee6b9800dc1dafb6fc5e571e7

pedrobaeza commented:
@gustitammam how are you handling additions? there are duplicated commits now. For example, I see 3 commits for "[MIG] web_m2x_options: Migration to 11.0", which should be only one.

gustitammam commented:
> @gustitammam how are you handling additions? there are duplicated commits now. For example, I see 3 commits for "[MIG] web_m2x_options: Migration to 11.0", which should be only one.

I'm doing this: `git format-patch --keep-subject --stdout origin/12.0..origin/11.0 -- $MODULE | git am -3 --keep`
Actually I also not sure about this way. How should I handle the change ?

pedrobaeza commented:
But each time you include a commit? That's why there are duplicated commits.

You should only cherry-pick the needed commit. Please clean the PR removing the duplicated commits and being sure no valid code is removed.

gustitammam commented:
> But each time you include a commit? That's why there are duplicated commits.
> 
> You should only cherry-pick the needed commit. Please clean the PR removing the duplicated commits and being sure no valid code is removed.

Please check, I was removing the duplicated commits with this steps:
- I reset to my first commit: https://github.com/Linksoft-ID/web/commit/5e8c3559c66855f989423575df0ac881b5b037ba
- Then I cherry-pick commit from this PR: https://github.com/OCA/web/pull/1118
- Then I cherry-pick this commit: https://github.com/OCA/web/commit/ddbbe81f4ea38c6685dd41efb5f11f1aeb8d0f9d

pedrobaeza commented:
Great, thanks. Now it looks correct.

@ernestotejeda please review this one.

ernestotejeda reviewed (commented):
if the option 'open' is set in true, when clicking on an element *A*, the associated resource will be opened in a dialog, but the color palette in the background is also shown. After closing the dialog, each time you click *A*, othe color palette is shown and not the dialog, unless you select another color for element A.
![wrong_web_m2x_options](https://user-images.githubusercontent.com/38267832/51637303-68918c00-1f29-11e9-97b8-7a3ff4b8d4f4.png)

ernestotejeda reviewed (commented):
The no_color_picker option doesn't seem to work

ernestotejeda reviewed (commented):
Generate README.rst file,  please

bealdav commented:
No news here ? Thanks

jcdrubay commented:
I tested for a few days and it worked well. nice :)

jcdrubay reviewed (approved):
I tested for few days and it worked nicely.

pedrobaeza commented:
@gustitammam can you check pending comment?

gustitammam commented:
> if the option 'open' is set in true, when clicking on an element _A_, the associated resource will be opened in a dialog, but the color palette in the background is also shown. After closing the dialog, each time you click _A_, othe color palette is shown and not the dialog, unless you select another color for element A.
> ![wrong_web_m2x_options](https://user-images.githubusercontent.com/38267832/51637303-68918c00-1f29-11e9-97b8-7a3ff4b8d4f4.png)

Honestly I never know about `web_m2x_options.open` option is exist, I used to use this options: `web_m2x_options.create`, `web_m2x_options.create_edit`, `web_m2x_options.m2o_dialog`.

@ernestotejeda I'm trying to reproduce your step, I set `web_m2x_options.open` to `True` but nothing happen when I click on many2many_tags. How can a popup show up in yours ?

@pedrobaeza Maybe I need some helps about this issue. It's quite difficult for me.

pedrobaeza commented:
@ernestotejeda please assist in this to @gustitammam

rven commented:
@gustitammam could you please do a `git rebase origin/12.0` on this PR?

ernestotejeda commented:
So, here are the steps

![peek 18-02-2019 10-29](https://user-images.githubusercontent.com/38267832/52993024-b1c9e400-33e0-11e9-81fa-01243fb270bb.gif)

bealdav commented:
Mergeable ?

gustitammam commented:
@ernestotejeda Thank you for the clue, `open` and `no_color_picker` is fixed now in https://github.com/OCA/web/pull/1086/commits/396e4f51ea6d4f2b55c45f2639313a6a8b33e13c, also I generate the README.rst in https://github.com/OCA/web/pull/1086/commits/a6e51e360103036fb6f70fde7ca8845eaf76bc9a

@rven done, it's rebased.

rven commented:
@gustitammam I think the rebase is not done correctly
first you have to fetch the changes of the origin and then you have to rebase these remote changes into your branch.
I'm getting conflicts on the latest 12.0 branch when applying your PR.
>>>>>>> Merge pull request #1187 from Tardo/pr_imp_sticky

git remote -v
origin  https://github.com/OCA/web.git (fetch)
origin  https://github.com/OCA/web.git (push)

git fetch origin              # Updates origin/12.0
git rebase origin/12.0   # Rebases current branch onto origin/12.0

then force push these changes.

Before you rebase you also have to fetch the remote branch for changes.

gustitammam commented:
@rven please try again.

ernestotejeda reviewed (approved):
It's working great

pedrobaeza commented:
Thanks for the contribution
